runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -569,6 +569,39 @@ TR::SymbolValidationManager::addMethodRecord(TR::MethodValidationRecord *record)
    return true;
    }
 
+bool
+TR::SymbolValidationManager::skipFieldRefClassRecord(
+   TR_OpaqueClassBlock *definingClass,
+   TR_OpaqueClassBlock *beholder,
+   uint32_t cpIndex)
+   {
+   // If the beholder refers to one of its own fields, or to a field of a
+   // well-known class, and if it does so by naming the class that declares the
+   // field (not a subclass), then no record is necessary.
+   if (definingClass == beholder || isWellKnownClass(definingClass))
```

Comment:
If the `definingClass` is equal to the `beholder`, do we need to go through this check? Is it possible for the `beholder` to name a subclass even though the field is declared by the `beholder`?

Reply:
Sure:
```
// Foo.java
class Foo {
    int x;
    int getBarX(Bar bar) { return bar.x; } // <---
}

// Bar.java
class Bar extends Foo {}
```